File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 9 and 10 March 2019/06. High Jump/06. High Jump.js
function highJump(arg) {
  let target = arg.shift()
  let current = target - 30
  let [count, fall] = [0, 0]

  while (current <= target) {
    count++
    let jump = arg.shift()
    current < jump ? (current += 5, fall = 0) : fall++
    if (fall > 2) return `Tihomir failed at ${current}cm after ${count} jumps.`
  }

  return `Tihomir succeeded, he jumped over ${target}cm after ${count} jumps.`
}

// console.log(highJump([231, 205, 212, 213, 228, 229, 230, 235]))
// console.log(highJump([250, 225, 224, 225, 228, 231, 235, 234, 235]))

File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 9 and 10 March 2019/06. High Jump/06. High Jump.py
target = int(input())
current = target - 30
count, fall = 0, 0

while current <= target:
      count += 1
      jump = int(input())

      if current < jump:
        current += 5
        fall = 0
      else:
         fall += 1

      if fall > 2:
            print(f'Tihomir failed at {current}cm after {count} jumps.')
            exit()

print(f'Tihomir succeeded, he jumped over {target}cm after {count} jumps.')


'''
231
205
212
213
228
229
230
235
'''
